pkg/service/nodes.go

```diff
@@ -49,51 +53,20 @@ func isNodeStatusInCondition(node v1.Node, condition v1.ConditionStatus) bool {
 	return false
 }
 
-func drainNode(kubectlPath, nodeName string, timeout, retryInterval int64, retryAttempts uint) error {
-	drainArgs := []string{"drain", nodeName, "--ignore-daemonsets=true", "--delete-local-data=true", "--force", "--grace-period=-1"}
-	drainCommand := kubectlPath
-
+func drainNode(kubeClient kubernetes.Interface, node *corev1.Node, timeout, retryInterval int64, retryAttempts uint) error {
 	if timeout == 0 {
 		log.Warn("skipping drain since timeout was set to 0")
 		return nil
 	}
 
-	err := runCommandWithContext(drainCommand, drainArgs, timeout, retryInterval, retryAttempts)
-	if err != nil {
-		if err.Error() == "command execution timed out" {
-			log.Warnf("failed to drain node %v, drain command timed-out", nodeName)
+	for retryAttempts > 0 {
+		err := drainNodeUtil(node, int(timeout), kubeClient)
+		if err != nil {
+			log.Errorf("failed to drain node %v  error: %v ", node.Name, err)
 			return err
 		}
```

Comment:
if err is nil, should we return directly?

Reply:
That's right. Modified it. Thanks for catching this!